Tools/CISupport/ews-build/events.py

```diff
@@ -315,26 +315,36 @@ def _get_commit_msg(self, repo, sha):
 
     @defer.inlineCallbacks
     def _get_pr_files(self, repo, number):
-        # Copied from https://github.com/buildbot/buildbot/blob/v2.10.5/master/buildbot/www/hooks/github.py to include added/modified/deleted
-        headers = {"User-Agent": "Buildbot"}
+        PER_PAGE = 100
+
+        headers = {}
         if self._token:
             headers["Authorization"] = "token " + self._token
 
-        url = "/repos/{}/pulls/{}/files".format(repo, number)
-        http = yield httpclientservice.HTTPClientService.getService(
-            self.master,
-            self.github_api_endpoint,
-            headers=headers,
-            debug=self.debug,
-            verify=self.verify,
-        )
-        res = yield http.get(url)
-        if 200 <= res.code < 300:
-            data = yield res.json()
-            return [self.file_with_status_sign(f) for f in data]
-
-        log.msg('Failed fetching PR files: response code {}'.format(res.code))
-        return []
+        page = 1
+        files = []
+        while page < 30:
+            response = yield TwistedAdditions.request(
+                url="{}/repos/{}/pulls/{}/files".format(self.github_api_endpoint, repo, number),
+                type=b'GET',
+                params=dict(
+                    per_page=PER_PAGE,
+                    page=page,
+                ), headers=headers,
+                logger=log.msg,
+            )
+            if not response or response.status_code // 100 != 2:
```

Comment:
should log something here for debugging

Reply:
We do, a `response` of `None` will end up with a log message from `TwistedAdditions.request` (usually means the request itself failed, and `TwistedAdditions.request` knows why that might have happened), and then we get an unconditional log outside of the loop when we have an empty list, which will include the response code (if there is one)